mktrayicon.c

```diff
@@ -256,10 +300,99 @@ gpointer watch_fifo(gpointer argv)
 #endif
 			break;
 		case 'm': /* menu */
-			fprintf(stderr, "Menu command not yet implemented\n");
-			if (param != NULL) {
-				free(param);
+			if (onmenu != NULL){
+				//destroy the previous menu
+				for (int i = 0; i < menusize; i++){
+					free(onmenu[i].name);
+					free(onmenu[i].action);
+					onmenu[i].name = NULL;
+					onmenu[i].action = NULL;
+				}
+				free(onmenu);
+				onmenu = NULL;
+				gtk_widget_destroy(menu);
+				menu = NULL;
 			}
+			
+			menusize = 0;
+
+			if (param != NULL && *param == '\0') {
+#ifdef DEBUG
+				printf("Removing onmenu handler\n");
+#endif
+				break;
+			}
+
+			// This block makes sure that the parameter after 'm' is well defined
+			int commas = 0;
+			int bars = 0;
+			if (len < 3){
+				break;
+			}
+			if (param[0] == ',' || param[0] == '|'){
+				printf("Action string cannot begin with ',' or '|'. Use '\\' to escape\n");
+				break;
+			}
+			if (param[len-2] == '|' && param[len-3] != '\\'){
+				printf("'|' can't be second of last. Use '\\' to escape\n");
+				break;
+			}
+			if (param[len-1] == ',' && param[len-2] != '\\'){
+				printf("',' can't be last. Use '\\' to escape\n");
+				break;
+			}
+			int i = 1; 
+			while (i < len - 1){
```

Comment:
Is there any particular reason you want this to be a `while` instead of a `for`?

Reply:
Not really. Pick of the moment I guess. I will change it.